lightning/src/routing/router.rs

```diff
@@ -1257,27 +1259,29 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 	// Step (9).
 	// Select the best route by lowest total fee.
 	drawn_routes.sort_by_key(|paths| paths.iter().map(|path| path.get_total_fee_paid_msat()).sum::<u64>());
-	let mut selected_paths = Vec::<Vec<RouteHop>>::new();
+	let mut selected_paths = Vec::<Vec<Result<RouteHop, LightningError>>>::new();
 	for payment_path in drawn_routes.first().unwrap() {
 		selected_paths.push(payment_path.hops.iter().map(|(payment_hop, node_features)| {
-			RouteHop {
-				pubkey: payment_hop.pubkey,
+			Ok(RouteHop {
+				pubkey: PublicKey::from_slice(&payment_hop.node_id.0).map_err(|_| LightningError{err: format!("Public key {:?} is invalid", &payment_hop.node_id), action: ErrorAction::IgnoreAndLog(Level::Trace)})?,
```

Comment:
With the suggested interface changes, I think we can revert the `Result` addition. Since we'd know the `NodeId` was constructed from a `PublicKey`, we can simply unwrap what comes  back from `PublicKey::from_slice`.